File: Java Lecture/ExamPrep/rotation.java
public class rotation{
    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};
        int size = arr.length;

        int k = 0;
        for(int i = 0; i < arr.length; i++){
            if(arr[i] < arr[k]){
                k = k + 1;
            }
        }

        for(int i = 0; i < arr.length; i++){
            System.out.print(arr[i] + " ");
        }
    }
}

File: Algorithms/Week_5/rotation.c
// This code finds number of rotations in a given sorted array
#include <stdio.h>
int find(int array[], int size) {

    int min_index = 0;

    for (int i = 1; i < size; i++) {
        if (array[i] < array[min_index]) {
            return i;
        }
    }
    
}
int check(int array[], int size) {
    for (int i = 1; i < size; i++) {
        if (array[i] < array[i - 1]) {
            return 0; 
        }
    }
    return 1; 
}
int main() {
    int nums[] = {5, 6, 1, 2, 3, 4};
    int size = sizeof(nums) / sizeof(nums[0]);
    if (check(nums,size)) {
        printf("rotation is 0\n");
    } else {
        int Counter = find(nums, size);
        printf("rotation is %d\n", Counter);
    }
    return 0;
}
